### PR TITLE
bug #1152: Fix for indexing small docs

### PR DESCRIPTION
**Pull request by marekjakimiuk1**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ X] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

It solves the problem with indexing small documents, where amount of text is less than section_overlap. In this case it's enought to check if  amount of text is less than max_section_length and return only one section without additional processing. 
This approach assumes that the section_overlap < max_section_length

```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
Index document whith amount of text less than max_section_length

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

Index document whith amount of text less than max_section_length

## What to Check
Verify that the following are valid
After indexing document whith amount of text less than max_section_length, check if document content has been added to the search index

### CODE REVIEW

File: scripts/prepdocslib/textsplitter.py [hunk 1]
```diff
@@ -42,59 +42,63 @@ def find_page(offset):
...
-        while start + self.section_overlap < length:
-            last_word = -1
-            end = start + self.max_section_length
-
-            if end > length:
-                end = length
-            else:
-                # Try to find the end of the sentence
+
+        if length <= self.max_section_length:
```
Comment (mattgotteiner) [inline scripts/prepdocslib/textsplitter.py:46]: I see - the diff is coming from the indentation after the else?

Comment (marekjakimiuk1) [reply scripts/prepdocslib/textsplitter.py:46]: Hi @mattgotteiner ,

I've added if statement befor while loop. For small documents where amount of text is less than section_overlap the condition: 'while start + self.section_overlap < length:' is always false same as the last if statement in the function ('if start + self.section_overlap < end:') so no section is return. For this solution, the assumptions is that section_overlap < max_section_length - but it seems to be obvious.
I've also reported bug related to usage of section_overlap in while loop: https://github.com/Azure-Samples/azure-search-openai-demo/issues/1153, but this is separate case.
Hope, this make it more clear.

Best regards,
Marek

Comment (mattgotteiner) [reply scripts/prepdocslib/textsplitter.py:46]: Thanks, this makes sense to me, I will manually test this and add some test cases. Appreciate it!

**Review by mattgotteiner — APPROVED** [review]

**Comment by mattgotteiner** [comment]
@marekjakimiuk1  thank you for your contribution!

**Comment by marekjakimiuk1** [comment]
> @marekjakimiuk1 thank you for your contribution! The PR has been edited slightly to reduce diff size and help with git blame in the future

@mattgotteiner Thank you too!

**Pull request merged**